src/main/java/org/gridsuite/study/server/StudyController.java

```diff
@@ -179,5 +181,24 @@ public StudyController(StudyService studyService) {
         return ResponseEntity.ok().contentType(MediaType.APPLICATION_JSON).body(studyMono);
 
     }
+
+    @GetMapping(value = "/studies/network-conversion/export/format")
+    @ApiOperation(value = "get the available export format", produces = "application/json")
+    @ApiResponses(value = {@ApiResponse(code = 200, message = "The available export format")})
+    public ResponseEntity<Mono<Collection<String>>> getExportFormats() {
+        Mono<Collection<String>> formatsMono = studyService.getExportFormats();
+        return ResponseEntity.ok().contentType(MediaType.APPLICATION_JSON).body(formatsMono);
+    }
+
+    @GetMapping(value = "/studies/network-conversion/{studyName}/export/{format}")
+    @ApiOperation(value = "export the study's network in the given format", produces = "application/json")
+    @ApiResponses(value = {@ApiResponse(code = 200, message = "The network in the given format")})
+    public ResponseEntity<byte[]> exportNetwork(@PathVariable("studyName") String studyName, @PathVariable("format") String format) {
+        Mono<ExportNetworkInfos> exportNetworkInfosMono = studyService.exportNetwork(studyName, format);
+        //if we don't block and the networkName is not ready yet the front will get some weird value for the filename
+        ExportNetworkInfos exportNetworkInfos = exportNetworkInfosMono.block();
```

Comment:
.block() should definitely be forbidden, you should try to understand why it is not working

Reply:
In this case where the header depends on the result of the webclient mono, you must use Mono<ResponseEntity<...>> instead of blocking